packages/app/src/cli/services/dev/processes/dev-session.test.ts

```diff
@@ -0,0 +1,157 @@
+import {setupDevSessionProcess, pushUpdatesForDevSession} from './dev-session.js'
+import {DeveloperPlatformClient} from '../../../utilities/developer-platform-client.js'
+import {AppLinkedInterface} from '../../../models/app/app.js'
+import {AppEventWatcher} from '../app-events/app-event-watcher.js'
+import {buildAppURLForWeb} from '../../../utilities/app/app-url.js'
+import {testAppLinked, testDeveloperPlatformClient, testWebhookExtensions} from '../../../models/app/app.test-data.js'
+import {formData} from '@shopify/cli-kit/node/http'
+import {describe, expect, test, vi, beforeEach} from 'vitest'
+import {AbortSignal} from '@shopify/cli-kit/node/abort'
+import {flushPromises} from '@shopify/cli-kit/node/promises'
+
+vi.mock('@shopify/cli-kit/node/fs')
```

Comment:
I don't think this mock is used, or archiver. It would be good if they weren't mocked

Reply:
Actually they are used, the upload process includes zipping the bundle folder, that's what is being mocked here with archiver (zipping) and fs (loading the zip file in memory)

Reply:
ok. I'd like us to start to limit this sort of mock -- I think we're best off exercising the file system in our tests -- working with files (and not arbitrary data structures) is _so_ fundamental to what the tool does